src/main/java/com/google/sps/servlets/TripServlet.java

```diff
@@ -61,13 +86,74 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     eventDoGet(response);
   }
 
+  /**
+   * Get user input.
+   * Generate directionsRequest from user input and parse optimized route.
+   * Create and store events in Datastore.
+   */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) 
       throws IOException {
     response.setContentType("application/json;");
 
+    Enumeration<String> params = request.getParameterNames();
+    String tripName = request.getParameter("inputTripName");
+    String origin = request.getParameter("inputDestination");
+    String startDate = request.getParameter("inputDayOfTravel");
+
+    // Print user input for now
+    response.getWriter().println("startDate: " + startDate);
+    response.getWriter().println("tripName: " + tripName);
+    response.getWriter().println("origin: " + origin);
+
+    // Save POIs to an array
+    List<String> pois = new ArrayList<>(); 
+    while (params.hasMoreElements()) {
+      String paramName = params.nextElement();
+      if (paramName.contains("poi")) {
+        String newPOI = request.getParameter(paramName);
+        pois.add(newPOI);
+      }
+    }
+    String[] poiStrings = new String[pois.size()]; 
+    poiStrings = pois.toArray(poiStrings); 
+
+    response.getWriter().println("original POI order: " + pois.toString());
+
+    //do post for maps
+    mapDoPost(request, response, origin, poiStrings);
+
     // do post for events
-    eventDoPost(request, response); 
+    eventDoPost(request, response, startDate, poiStrings); 
+  }
+
+  /**
+   * Make servlet cleaner
+   * Generate directionsRequest from user input and parse optimized route.
+   */
+  private void mapDoPost(HttpServletRequest request, HttpServletResponse response, 
```

Comment:
From what I remember the reason you aren't storing the objects in datastore is because its such a big problem/hassle. Are you getting the frontend info from the response writer? We were talking about putting entities in datastore that kept track of the order, I said to maybe do this after my CL went through but I think you should do it first. This will allow me to access the correct order.